client.go

```diff
@@ -182,32 +182,35 @@ func (cl *Client) announceKey() int32 {
 	return int32(binary.BigEndian.Uint32(cl.peerID[16:20]))
 }
 
+// Construct a bare minimum Client. *ClientConfig argument must not be nil.
+func makeClient(cfg *ClientConfig) (cl Client) {
+	cl.config = cfg
+	cl.dopplegangerAddrs = make(map[string]struct{})
+	cl.torrents = make(map[metainfo.Hash]*Torrent)
+	cl.dialRateLimiter = rate.NewLimiter(10, 10)
+	cl.activeAnnounceLimiter.SlotsPerKey = 2
+	
+	cl.ipBlockList = cfg.IPBlocklist
+	return
+}
+
 func NewClient(cfg *ClientConfig) (cl *Client, err error) {
 	if cfg == nil {
 		cfg = NewDefaultClientConfig()
 		cfg.ListenPort = 0
 	}
-	defer func() {
-		if err != nil {
-			cl = nil
-		}
-	}()
-	cl = &Client{
-		config:            cfg,
-		dopplegangerAddrs: make(map[string]struct{}),
-		torrents:          make(map[metainfo.Hash]*Torrent),
-		dialRateLimiter:   rate.NewLimiter(10, 10),
-	}
-	cl.activeAnnounceLimiter.SlotsPerKey = 2
+	client := makeClient(cfg)
+	cl = &client
 	go cl.acceptLimitClearer()
 	cl.initLogger()
 	defer func() {
-		if err == nil {
-			return
+		if err != nil {
+			cl.Close()
+			cl = nil
 		}
-		cl.Close()
 	}()
 	cl.event.L = cl.locker()
```

Comment:
Shouldn't this be in makeClient?

Reply:
I was getting a build issue when I moved it in makeClient for some reason. There was a panic about a lock / RWMutex ... ah I don't remember exactly but it was frustrating. Anyway it went away right when I moved `cl.event.L = cl.locker()` back to its original position. Probably a deeper issue at hand. I can move it back to see if the issue still exists. Wanna try that?

Reply:
I'll just do it, and if it comes up -- have a look lol.